Dragonflight/ShamanElemental.lua

```diff
@@ -935,15 +996,34 @@ spec:RegisterAbilities( {
 
             -- 4 MS per target, direct.
             -- 3 MS per target, overload.
-
-            gain( ( buff.stormkeeper.up and 7 or 4 ) * min( 5, active_enemies ), "maelstrom" )
+			-- stormkeeper guarantees overload on every target hit
+			-- power of the maelstrom guarantees 1 extra overload on the initial target
+			-- surge of power adds 1 extra target to total potential enemies hit
+
+            gain( ( buff.stormkeeper.up and 4 + ( min( (buff.surge_of_power.up and 6 or 5),active_enemies ) * 3) or 4 ) * min( (buff.surge_of_power.up and 6 or 5), active_enemies ), "maelstrom" )
+			if buff.power_of_the_maelstrom.up then
+				gain( 3 * min( ( buff.surge_of_power.up and 6 or 5 ), active_enemies ), "maelstrom")
+			end
+			
             removeStack( "stormkeeper" )
+			removeStack( "power_of_the_maelstrom" )
+			removeBuff( "surge_of_power" )
 
             if pet.storm_elemental.up then
                 addStack( "wind_gust", nil, 1 )
             end
-
-
+	
+			if talent.flash_of_lightning.enabled then
+				--TODO: reduce cd of all nature spells
```

Comment:
You listed all the spells, that's also great.  We can drop their names in a table and `reduceCooldown` by the proper amount when appropriate.

Reply:
Done.